core/trino-main/src/main/java/io/trino/sql/planner/planprinter/ValuePrinter.java

```diff
@@ -59,4 +59,8 @@ public String castToVarcharOrFail(Type type, Object value)
         Slice coerced = (Slice) new InterpretedFunctionInvoker(functionManager).invoke(coercion, session.toConnectorSession(), value);
         return coerced.toStringUtf8();
     }
+
+    public Session getSession() {
```

Comment:
Session shouldn't be exposed from here, take Session as input to `formatFragment` if needed